Task5/src/controller/auth.controller.ts

```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from 'express';
+import { AuthService } from './../service';
+import { constructResponse, contructErrorResponse } from "../util";
+import { userAuthSchema } from '../schema';
+import jwt from 'jsonwebtoken';
+import config from './../config/secret.json';
+
+export class AuthController {
+    private authService: AuthService;
+
+    constructor() {
+        this.authService = new AuthService();
+    }
+
+    public authenticate = (req: Request, res: Response) => {
+        const { error } = userAuthSchema(req);
+
+        if (error) {
+            contructErrorResponse(req, res, error, 422);
+        }
+        else {
+            const { name, password } = req.body;
+
+            this.authService
+                .authenticate(name, password)
+                .then((user: any) => {
+                    if (user) {
+                        // create a jwt token that is valid for 7 days
+                        const token = jwt.sign({ sub: user.id }, config.secret, { expiresIn: '7d' });
+                        return constructResponse(res, token, 'Authenticated', 200);
+                    }
+                })
+                .catch((err: any) => contructErrorResponse(req, res, err, 400, false));
```

Comment:
Error code should be `404`, don't you think ? This service is trying to find user and in case the record is not available, then 'Not Found' should be sent.

Reply:
Also, in case of incorrect token, error code should be `403`. That condition is not handled here.

Reply:
yeah I missed to handle `404`, added